### DISCUSSION TITLE
Mesh error

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27395] [Q&A Meshing]
I have stitched two meshes together from previous simulations. Below is how the mesh was generated in the previous simulation:
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 150
  ny = 150
#  nz = 2
  xmin = 0
  xmax = 300
  ymin = 0
  ymax = 300
  zmin = 0
  zmax = 0
#  elem_type = QUAD4
[]

Below is how I stitched together:
[Mesh]
  [./fmg_left]
    type = FileMeshGenerator
    file = my_kks_noload_110_out.e-s172
  []

  [./fmg_right]
    type = FileMeshGenerator
    file = my_kks_noload_one8_out.e-s172
  []

  [./smg]
    type = StitchedMeshGenerator
    inputs = 'fmg_left fmg_right'
    clear_stitched_boundary_ids = true
    stitch_boundaries_pairs = 'right left'
  []
[]

Yet I get this as the output when I test only the mesh:
[emmanuel324@tinkercliffs1 problems]$ ../darcy_thermo_mech-opt -i bicrystal_kk
s_mesh.i --mesh-only
In UnstructuredMesh::stitch_meshes:
This mesh has 182 nodes on boundary `right' (1).
Other mesh has 174 nodes on boundary `' (7).
Minimum edge length on both surfaces is 1.
In UnstructuredMesh::stitch_meshes:
Found 0 matching nodes.

 Mesh Information:
  elem_dimensions()={2}
  spatial_dimension()=2
  n_nodes()=75841
    n_local_nodes()=75841
  n_elem()=71685
    n_local_elem()=71685
    n_active_elem()=71685
  n_subdomains()=1
  n_elemsets()=0
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0
  is_prepared()=true
  is_replicated()=true

 Mesh Bounding Box:
  Minimum: (x,y,z)=(       0,        0,        0)
  Maximum: (x,y,z)=(     300,      300,        0)
  Delta:   (x,y,z)=(     300,      300,        0)

 Mesh Element Type(s):
  QUAD4

 Mesh Nodesets:
  Nodeset 0 (bottom), 151 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 1 (right), 182 nodes
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 2 (top), 223 nodes
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 3 (left), 179 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 4 (bottom), 183 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 5 (right), 226 nodes
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 6 (top), 178 nodes
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 7 (left), 174 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)

 Mesh Sidesets:
  Sideset 0 (bottom), 150 sides (EDGE2), 150 elems (QUAD4), 151 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 1 (right), 181 sides (EDGE2), 181 elems (QUAD4), 182 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 2 (top), 222 sides (EDGE2), 222 elems (QUAD4), 223 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 3 (left), 178 sides (EDGE2), 178 elems (QUAD4), 179 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 4 (bottom), 182 sides (EDGE2), 182 elems (QUAD4), 183 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 5 (right), 225 sides (EDGE2), 225 elems (QUAD4), 226 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 6 (top), 177 sides (EDGE2), 177 elems (QUAD4), 178 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 7 (left), 173 sides (EDGE2), 173 elems (QUAD4), 174 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
 Mesh Edgesets:
  None
 Mesh Subdomains:
  Subdomain 0: 71685 elems (QUAD4, 71685 active), 75841 active nodes
   Volume: 180000
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,      300,        0)
  Global mesh volume = 180000
```
And the .e file does not give anything meaningful when I want to view it. @dschwen @GiudGiud

**Comment by GiudGiud** [comment]
```
[Mesh]
  [./fmg_left]
    type = FileMeshGenerator
    file = my_kks_noload_110_out.e-s172
  []

  [./fmg_right]
    type = FileMeshGenerator
    file = my_kks_noload_one8_out.e-s172
  []

  [./smg]
    type = StitchedMeshGenerator
    inputs = 'fmg_left fmg_right'
    clear_stitched_boundary_ids = true
    stitch_boundaries_pairs = 'right left'
  []
[]
```
these files come from adaptivity dont they?
if the left and right boundary are at different levels of refinement, they are not going to be stitchable properly on a node-to-node basis.

I suspect 'right' and 'left' do not align spatially at all either, since you get 0 stitched nodes. 
you need to use a TransformMeshGenerator to make the two sidesets super-impose

**Reply by Emmanuel-324** [reply]
Yes, they do come from an adaptive mesh.

**Reply by Emmanuel-324** [reply]
I get this error message:

*** ERROR ***
A 'TransformMeshGenerator' is not a registered object.
If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

**Reply by GiudGiud** [reply]
https://mooseframework.inl.gov/source/meshgenerators/TransformGenerator.html
please feel free to use the documentation :D

**Reply by Emmanuel-324** [reply]
I am not getting any information close to my problem as I am trying to also stitch two meshes together. I do not know which one to transform, scale, or translate. The example input scripts are not helping

**Reply by GiudGiud** [reply]
the two sidesets you are trying to stitch must overlap. You need to translate one if they do not currently overlap

**Reply by Emmanuel-324** [reply]
I  get this error when I try to translate. Maybe I do not understand the translation well.


*** ERROR ***
Your MeshGenerator tree contains multiple possible generator outputs :
"translate_mesh and one or more of the following from an independent set: "fmg_left, fmg_righ
t"

Below is the input script:
[Mesh]
  [./fmg_left]
    type = FileMeshGenerator
    file = my_kks_noload_110_out.e-s172
  []
  
  [./fmg_right]
    type = FileMeshGenerator
    file = my_kks_noload_one8_out.e-s172
  []
  
  [translate_mesh]
    type = TransformGenerator
    input = fmg_left
    transform = translate
    vector_value = '1 2 0'
  []
[]
This may be due to a missing dependency or may be intentional. Please either
- check that all the mesh generators are connected as a tree and culminate in a single final 
mesh. Having one wrong 'input=mg' parameter is the most common error
- add additional dependencies to remove the ambiguity if you are using a user-built MeshGener
ator
- if you intend to execute a subset of the defined generators (uncommon), select the final Me
shGenerator in the [Mesh] block with the "final_generator" parameter.

**Reply by GiudGiud** [reply]
you need to have a tree with a single output for the mesh generators
Add the stitch mesh generator, this joins the two branches

**Reply by Emmanuel-324** [reply]
I get the same error:
** ERROR ***
Your MeshGenerator tree contains multiple possible generator outputs :
"translate_mesh and one or more of the following from an independent set: "fmg_left, smg, translate_mesh"
This may be due to a missing dependency or may be intentional. Please either
- check that all the mesh generators are connected as a tree and culminate in a single final mesh. Having one wrong 'input=mg' parameter is the most common error
- add additional dependencies to remove the ambiguity if you are using a user-built MeshGenerator
- if you intend to execute a subset of the defined generators (uncommon), select the final MeshGenerator in the [Mesh] block with the "final_generator" parameter.

input script:
[Mesh]
  [./fmg_left]
    type = FileMeshGenerator
    file = my_kks_noload_110_out.e-s172
  []
  
  [./fmg_right]
    type = FileMeshGenerator
    file = my_kks_noload_one8_out.e-s172
  []
  
  [translate_mesh]
    type = TransformGenerator
    input = fmg_right
    transform = translate
    vector_value = '1 2 0'
  []

  [./smg]
    type = StitchedMeshGenerator
    inputs = 'fmg_left fmg_right'
    clear_stitched_boundary_ids = true
    stitch_boundaries_pairs = 'right left'
  []
[]

**Reply by GiudGiud** [reply]
do this
```
[./smg]
type = StitchedMeshGenerator
inputs = 'fmg_left translate_mesh'
clear_stitched_boundary_ids = true
stitch_boundaries_pairs = 'right left'
[]
```

**Reply by Emmanuel-324** [reply]
Still 0 matching nodes

In UnstructuredMesh::stitch_meshes:
This mesh has 182 nodes on boundary `right' (1).
Other mesh has 174 nodes on boundary `' (7).
Minimum edge length on both surfaces is 1.
In UnstructuredMesh::stitch_meshes:
Found 0 matching nodes.

 Mesh Information:
  elem_dimensions()={2}
  spatial_dimension()=2
  n_nodes()=75841
    n_local_nodes()=75841
  n_elem()=71685
    n_local_elem()=71685
    n_active_elem()=71685
  n_subdomains()=1
  n_elemsets()=0
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0
  is_prepared()=true
  is_replicated()=true

 Mesh Bounding Box:
  Minimum: (x,y,z)=(       0,        0,        0)
  Maximum: (x,y,z)=(     301,      302,        0)
  Delta:   (x,y,z)=(     301,      302,        0)

 Mesh Element Type(s):
  QUAD4

 Mesh Nodesets:
  Nodeset 0 (bottom), 151 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 1 (right), 182 nodes
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 2 (top), 223 nodes
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 3 (left), 179 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 4 (bottom), 183 nodes
   Bounding box minimum: (x,y,z)=(       1,        2,        0)
   Bounding box maximum: (x,y,z)=(     301,        2,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 5 (right), 226 nodes
   Bounding box minimum: (x,y,z)=(     301,        2,        0)
   Bounding box maximum: (x,y,z)=(     301,      302,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Nodeset 6 (top), 178 nodes
   Bounding box minimum: (x,y,z)=(       1,      302,        0)
   Bounding box maximum: (x,y,z)=(     301,      302,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Nodeset 7 (left), 174 nodes
   Bounding box minimum: (x,y,z)=(       1,        2,        0)
   Bounding box maximum: (x,y,z)=(       1,      302,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)

 Mesh Sidesets:
  Sideset 0 (bottom), 150 sides (EDGE2), 150 elems (QUAD4), 151 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,        0,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 1 (right), 181 sides (EDGE2), 181 elems (QUAD4), 182 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(     300,        0,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 2 (top), 222 sides (EDGE2), 222 elems (QUAD4), 223 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,      300,        0)
   Bounding box maximum: (x,y,z)=(     300,      300,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 3 (left), 178 sides (EDGE2), 178 elems (QUAD4), 179 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,      300,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 4 (bottom), 182 sides (EDGE2), 182 elems (QUAD4), 183 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       1,        2,        0)
   Bounding box maximum: (x,y,z)=(     301,        2,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 5 (right), 225 sides (EDGE2), 225 elems (QUAD4), 226 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(     301,        2,        0)
   Bounding box maximum: (x,y,z)=(     301,      302,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
  Sideset 6 (top), 177 sides (EDGE2), 177 elems (QUAD4), 178 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       1,      302,        0)
   Bounding box maximum: (x,y,z)=(     301,      302,        0)
   Bounding box delta: (x,y,z)=(     300,        0,        0)
  Sideset 7 (left), 173 sides (EDGE2), 173 elems (QUAD4), 174 nodes
   Side volume: 300
   Bounding box minimum: (x,y,z)=(       1,        2,        0)
   Bounding box maximum: (x,y,z)=(       1,      302,        0)
   Bounding box delta: (x,y,z)=(       0,      300,        0)
 Mesh Edgesets:
  None
 Mesh Subdomains:
  Subdomain 0: 71685 elems (QUAD4, 71685 active), 75841 active nodes
   Volume: 180000
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(     301,      302,        0)
   Bounding box delta: (x,y,z)=(     301,      302,        0)
  Global mesh volume = 180000

**Reply by GiudGiud** [reply]
that s unfortunate
do you mind posting a picture of the 2 meshes separately and the two meshes together? they should all be viewable in paraivew

**Reply by Emmanuel-324** [reply]
![Screenshot 2024-04-17 164407](https://github.com/idaholab/moose/assets/144969366/cbd8cba1-587a-4f0c-83e3-f1778c6dd408)
![Screenshot 2024-04-17 164454](https://github.com/idaholab/moose/assets/144969366/746d9676-2791-4e1b-bf22-f0ed11a58086)
![Screenshot 2024-04-17 164540](https://github.com/idaholab/moose/assets/144969366/91150819-460c-44f8-978b-d5a2034798b7)

**Reply by GiudGiud** [reply]
can you plot with surface with edges? so we see the mesh
also the 3rd picture is still both separated, not the output of the script that did not stitch well

**Reply by Emmanuel-324** [reply]
Below is the output when they're together:
![image](https://github.com/idaholab/moose/assets/144969366/dc951e16-3914-468d-b81a-b3bb4e2b939c)
![image](https://github.com/idaholab/moose/assets/144969366/43099f4b-23a8-4bb3-9195-828fc4f34453)
![Screenshot 2024-04-17 170407](https://github.com/idaholab/moose/assets/144969366/6bdb5f8c-da19-4eb4-a33e-0302929baf06)

**Reply by Emmanuel-324** [reply]
I tried copying the error output when they're together in pararview but it comes out as a picture. I will send it again

**Reply by Emmanuel-324** [reply]
For some weird reasons I am not able to copy the error message

**Reply by Emmanuel-324** [reply]
![Screenshot 2024-04-17 175817](https://github.com/idaholab/moose/assets/144969366/adc2b2d0-53c8-4ae0-8f63-420036bd7bcb)
![Screenshot 2024-04-17 175832](https://github.com/idaholab/moose/assets/144969366/0b98a9c6-cc9a-4652-83bd-ca4c3b4854be)
This is the output message from paraview

**Reply by GiudGiud** [reply]
what version is your paraview? I think latest paraview can handle that better

**Reply by Emmanuel-324** [reply]
5.12.0

**Reply by GiudGiud** [reply]
ok same.
you ll need to use the RenameBoundaryGenerator to differentiate the sidesets in both meshes then to make this work

**Reply by Emmanuel-324** [reply]
How do I get the boundary names/id for my old and new boundaries? Below is the input script I am trying to use:
```
[Mesh]
  [./fmg_left]
    type = FileMeshGenerator
    file = my_kks_noload_110_out.e-s172
  []
  
  [./fmg_right]
    type = FileMeshGenerator
    file = my_kks_noload_one8_out.e-s172
  []
  
  [rename]
    type = RenameBoundaryGenerator
    input = fmg_right
    old_boundary = 'meaningless 5'
    new_boundary = 'interior exterior'
  []    

  [./smg]
    type = StitchedMeshGenerator
    inputs = 'fmg_left rename'
    clear_stitched_boundary_ids = true
    stitch_boundaries_pairs = 'right left'
  []
[]
```

**Reply by GiudGiud** [reply]
use show_info = true in each generator to get that information for the output of each generator

**Reply by Emmanuel-324** [reply]
This is the information I get but can't actually figure out what I need
```
python-stack) [emmanuel324@tinkercliffs1 problems]$ ../darcy_thermo_mech-opt -i bicrystal_kks_mesh.i --mesh-only
FileMeshGenerator 'fmg_left':  Mesh Information:
FileMeshGenerator 'fmg_left':   elem_dimensions()={2}
FileMeshGenerator 'fmg_left':   spatial_dimension()=2
FileMeshGenerator 'fmg_left':   n_nodes()=37590
FileMeshGenerator 'fmg_left':     n_local_nodes()=37590
FileMeshGenerator 'fmg_left':   n_elem()=35544
FileMeshGenerator 'fmg_left':     n_local_elem()=35544
FileMeshGenerator 'fmg_left':     n_active_elem()=35544
FileMeshGenerator 'fmg_left':   n_subdomains()=1
FileMeshGenerator 'fmg_left':   n_elemsets()=0
FileMeshGenerator 'fmg_left':   n_partitions()=1
FileMeshGenerator 'fmg_left':   n_processors()=1
FileMeshGenerator 'fmg_left':   n_threads()=1
FileMeshGenerator 'fmg_left':   processor_id()=0
FileMeshGenerator 'fmg_left':   is_prepared()=true
FileMeshGenerator 'fmg_left':   is_replicated()=true
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Bounding Box:
FileMeshGenerator 'fmg_left':   Minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':   Maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':   Delta:   (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Element Type(s):
FileMeshGenerator 'fmg_left':   QUAD4
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Nodesets:
FileMeshGenerator 'fmg_left':   Nodeset 0 (bottom), 151 nodes
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':   Nodeset 1 (right), 182 nodes
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':   Nodeset 2 (top), 223 nodes
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':   Nodeset 3 (left), 179 nodes
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Sidesets:
FileMeshGenerator 'fmg_left':   Sideset 0 (bottom), 150 sides (EDGE2), 150 elems (QUAD4), 151 nodes
FileMeshGenerator 'fmg_left':    Side volume: 300
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':   Sideset 1 (right), 181 sides (EDGE2), 181 elems (QUAD4), 182 nodes
FileMeshGenerator 'fmg_left':    Side volume: 300
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':   Sideset 2 (top), 222 sides (EDGE2), 222 elems (QUAD4), 223 nodes
FileMeshGenerator 'fmg_left':    Side volume: 300
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_left':   Sideset 3 (left), 178 sides (EDGE2), 178 elems (QUAD4), 179 nodes
FileMeshGenerator 'fmg_left':    Side volume: 300
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Edgesets:
FileMeshGenerator 'fmg_left':   None
FileMeshGenerator 'fmg_left': 
FileMeshGenerator 'fmg_left':  Mesh Subdomains:
FileMeshGenerator 'fmg_left':   Subdomain 0: 35544 elems (QUAD4, 35544 active), 37590 active nodes
FileMeshGenerator 'fmg_left':    Volume: 90000
FileMeshGenerator 'fmg_left':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_left':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':    Bounding box delta: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_left':   Global mesh volume = 90000
FileMeshGenerator 'fmg_right':  Mesh Information:
FileMeshGenerator 'fmg_right':   elem_dimensions()={2}
FileMeshGenerator 'fmg_right':   spatial_dimension()=2
FileMeshGenerator 'fmg_right':   n_nodes()=38251
FileMeshGenerator 'fmg_right':     n_local_nodes()=38251
FileMeshGenerator 'fmg_right':   n_elem()=36141
FileMeshGenerator 'fmg_right':     n_local_elem()=36141
FileMeshGenerator 'fmg_right':     n_active_elem()=36141
FileMeshGenerator 'fmg_right':   n_subdomains()=1
FileMeshGenerator 'fmg_right':   n_elemsets()=0
FileMeshGenerator 'fmg_right':   n_partitions()=1
FileMeshGenerator 'fmg_right':   n_processors()=1
FileMeshGenerator 'fmg_right':   n_threads()=1
FileMeshGenerator 'fmg_right':   processor_id()=0
FileMeshGenerator 'fmg_right':   is_prepared()=true
FileMeshGenerator 'fmg_right':   is_replicated()=true
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Bounding Box:
FileMeshGenerator 'fmg_right':   Minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':   Maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':   Delta:   (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Element Type(s):
FileMeshGenerator 'fmg_right':   QUAD4
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Nodesets:
FileMeshGenerator 'fmg_right':   Nodeset 0 (bottom), 183 nodes
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':   Nodeset 1 (right), 226 nodes
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':   Nodeset 2 (top), 178 nodes
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':   Nodeset 3 (left), 174 nodes
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Sidesets:
FileMeshGenerator 'fmg_right':   Sideset 0 (bottom), 182 sides (EDGE2), 182 elems (QUAD4), 183 nodes
FileMeshGenerator 'fmg_right':    Side volume: 300
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':   Sideset 1 (right), 225 sides (EDGE2), 225 elems (QUAD4), 226 nodes
FileMeshGenerator 'fmg_right':    Side volume: 300
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':   Sideset 2 (top), 177 sides (EDGE2), 177 elems (QUAD4), 178 nodes
FileMeshGenerator 'fmg_right':    Side volume: 300
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(     300,        0,        0)
FileMeshGenerator 'fmg_right':   Sideset 3 (left), 173 sides (EDGE2), 173 elems (QUAD4), 174 nodes
FileMeshGenerator 'fmg_right':    Side volume: 300
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(       0,      300,        0)
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Edgesets:
FileMeshGenerator 'fmg_right':   None
FileMeshGenerator 'fmg_right': 
FileMeshGenerator 'fmg_right':  Mesh Subdomains:
FileMeshGenerator 'fmg_right':   Subdomain 0: 36141 elems (QUAD4, 36141 active), 38251 active nodes
FileMeshGenerator 'fmg_right':    Volume: 90000
FileMeshGenerator 'fmg_right':    Bounding box minimum: (x,y,z)=(       0,        0,        0)
FileMeshGenerator 'fmg_right':    Bounding box maximum: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':    Bounding box delta: (x,y,z)=(     300,      300,        0)
FileMeshGenerator 'fmg_right':   Global mesh volume = 90000
```

**Reply by GiudGiud** [reply]
look for the info below this line for each `FileMeshGenerator 'fmg_left':  Mesh Sidesets:`

then use that info to rename the output of one mesh generator

**Reply by Emmanuel-324** [reply]
For your information, I am only trying to make the mesh work before I use it. So I am running --mesh-only all this while so when you mention output do you mean the output block?

**Reply by GiudGiud** [reply]
by output I mean the mesh that comes out of the mesh generator

**Comment by GiudGiud** [comment]
there is also a parameter called prevent_boundary_overlaps in the stichedMeshGenerator
this will merge the boundaries with the same names if you use it with 'false' instead of the default.

keep in mind merging boundaries is usually not the right thing to do, but it will load in paraivew
